### PR TITLE
feat(cli/test): `deno test --clean`

### LINKED ISSUE #23491: `deno test --coverage` should clear the coverage directory first
Version: Deno 1.42.4

I just noticed that I had hundreds of thousands of files in `coverage/` directory of my local checkout of `deno_std`. I've been running tests before the release for the past 6 months and accumulated a lot of files. Since this directory should be git ignored anyway, it probably makes sense to clear the directory when the command is invoked.

### PR DESCRIPTION
**Pull request by iuioiua**
The result of the call is ignored as it throws even when the directory does not exist.

Closes #23491

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Please add an integration test

**Comment by lucacasonato** [comment]
This has the downside that you can not run multiple `deno test` calls in parallel, or even in series unless you specify a different directories.

**Comment by iuioiua** [comment]
I think it's a fair trade-off, as long as the behaviour is clearly documented. Perhaps, if we receive too many complaints, which we may not, we can add a `--no-clean`, or similar, flag.

For context, I recently waited for ~5 minutes on my M2 MacBook to delete `/coverage` for `std`.

**Comment by lucacasonato** [comment]
The point is, as is, this PR breaks existing workflows

**Comment by iuioiua** [comment]
We could make this behaviour opt-in via `--clean-cov-dir` (or similar) and then at some point decide whether we make this default behaviour in Deno 2.

**Comment by iuioiua** [comment]
I've changed my mind. I think this should be opt-in.

File: cli/args/flags.rs [hunk 1]
```diff
@@ -2459,6 +2460,14 @@ Directory arguments are expanded to all contained files matching the glob
...
         .help("Collect coverage profile data into DIR. If DIR is not specified, it uses 'coverage/'."),
     )
+    .arg(
+      Arg::new("clean")
+        .long("clean")
+        .help("Empty the coverage profile data directory before running tests.
+        
+Note: running multiple `deno test --clean` calls in series or parallel for the same coverage directory may cause race conditions.")
+        .action(ArgAction::SetTrue),
+    )
```
Comment (bartlomieju) [inline cli/args/flags.rs:2470]: This makes sense, but I'm wondering if we should reword this to say 
```
"Empty any temporary files created by `deno test` before running the tests. Currently this includes coverage data, but may change in the future."
```
Thoughts?

Comment (iuioiua) [reply cli/args/flags.rs:2470]: I feel like saying it might change in the future might cause some confusion. IMO, how it is now is fine. At most, we could do "Empty the **temporary** coverage profile data..."

Comment (bartlomieju) [reply cli/args/flags.rs:2632]: Okay, that seems reasonable, please update the message

**Review by bartlomieju — APPROVED** [review]

**Comment by iuioiua** [comment]
Should this only work with `--coverage`?
